<launch>

  <node pkg="lidar_align" name="lidar_align" type="lidar_align">
    <param name="input_bag_path" value="/home/z/Datasets/lidar/2018-09-11-20-51-12.bag" />
    <param name="output_pointcloud_path" value="/home/z/Desktop/lidar.ply" />
    <param name="output_calibration_path" value="/home/z/Desktop/calib.txt" />

    <param name="use_n_scans" value="1000" />
    <param name="local" value="true" />
    <param name="time_cal" value="true" />
    <param name="motion_compensation" value="true" />
    <param name="clockwise_lidar" value="false" />
    <rosparam param="inital_guess">[0.0, 0.0, 0.0, -3.0, 0.0, 0.0, 0.0]</rosparam>
    <rosparam param="range">[0.3, 0.3, 0.3, 0.5, 0.5, 0.5, 0.1]</rosparam>
    <param name="max_evals" value="500" />
    <param name="xtol" value="0.0000001" />
    
    <param name="min_point_distance" value="2" />
    <param name="max_point_distance" value="10" />
    <param name="keep_points_ratio" value="0.01" />
    <param name="max_per_scan_points" value="10000" />

    <param name="knn_batch_size" value="5000" />
    <param name="knn_k" value="1" />
    <param name="knn_max_dist" value="0.1" />
  </node>

</launch>